src/parser/mod.rs

```diff
@@ -3894,9 +3894,15 @@ impl<'a> Parser<'a> {
         match next_token.token {
             Token::Word(w) if w.keyword == Keyword::PRIMARY || w.keyword == Keyword::UNIQUE => {
                 let is_primary = w.keyword == Keyword::PRIMARY;
-                if is_primary {
+                let name = if is_primary {
```

Comment:
I wonder if this code could be simpler if it didn't check for `is_primary` and just always did the `self.maybe_parse` clause?

Reply:
Sorry, I have no idea how to optimize this code...      T T